Source/WebCore/page/Navigation.cpp

```diff
@@ -178,9 +178,9 @@ Navigation::Result Navigation::reload(ReloadOptions&& options, Ref<DeferredPromi
 }
 
 // https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-navigate
-Navigation::Result Navigation::navigate(ScriptExecutionContext& scriptExecutionContext, const String& url, NavigateOptions&& options, Ref<DeferredPromise>&& committed, Ref<DeferredPromise>&& finished)
+Navigation::Result Navigation::navigate(const String& url, NavigateOptions&& options, Ref<DeferredPromise>&& committed, Ref<DeferredPromise>&& finished)
 {
-    auto currentURL = scriptExecutionContext.url();
+    auto currentURL = scriptExecutionContext()->url();
```

Comment:
The code change looks right to me. To get the resolve the URL, you're using the *relevant* script execution context (instead of the *current* one), which is what the spec [1] says:
> Let urlRecord be the result of encoding-parsing a URL given url, relative to this's relevant settings object.

[1] https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-navigation-navigate